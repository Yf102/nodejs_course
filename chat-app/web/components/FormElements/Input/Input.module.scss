.text {
    background: red;
    color: #fff;
    border-radius: 0 0 5px 5px;
    display: flex;
    align-items: center;
    justify-content: center;
    font-size: 14px;
    overflow: hidden;
}

.input-style {
  @apply bg-white border-0;
  @apply rounded-md outline-none pl-2;
  @apply focus:outline-none mb-2 text-black;
  max-height: 50px;
  height: 50px;
  width: calc(100% - 115px);
}