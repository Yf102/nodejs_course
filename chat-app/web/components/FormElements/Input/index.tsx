import { DetailedHTMLProps, forwardRef, InputHTMLAttributes } from 'react'
import styles from './Input.module.scss'
import cn from 'classnames'

type InputProps = DetailedHTMLProps<
  InputHTMLAttributes<HTMLInputElement>,
  HTMLInputElement
>

type InputCustomProps = {
  className?: string
}

const Input = forwardRef<HTMLInputElement, InputProps & InputCustomProps>(
  (props, ref) => {
    const { className } = props

    return (
      <input
        data-testid='input-element'
        ref={ref}
        {...props}
        className={cn(styles['input-style'], className)}
      />
    )
  }
)

Input.displayName = 'Input'

export default Input
