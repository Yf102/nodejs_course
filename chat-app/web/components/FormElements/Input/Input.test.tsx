import React from 'react'
import { render, screen } from 'tests/test-utils'
import Input from 'components/FormElements/Input/index'

describe('Input', () => {
  it('It renders Input', async () => {
    const { asFragment } = render(<Input placeholder='23.351723' />)
    expect(screen.getByTestId('input-element')).toBeTruthy()
    expect(asFragment()).toMatchSnapshot()
  })
})
