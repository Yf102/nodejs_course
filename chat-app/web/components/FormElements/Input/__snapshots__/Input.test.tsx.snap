// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input It renders Input 1`] = `
<DocumentFragment>
  <input
    class="input-style"
    data-testid="input-element"
    placeholder="23.351723"
  />
</DocumentFragment>
`;
